### DISCUSSION TITLE
Execution steps

### DISCUSSION THREAD

**Discussion by CVasilakis** [007revad/Synology_enable_M2_volume#96] [Q&A] [answered]
Hello,
Thanks a lot for your great work on this project.
I'm preparing to run the script and want to double-check that i have understand the required steps right.

1. Execute:  `sudo -i <PATH_TO_SCRIPT>/syno_enable_m2_volume.sh`
2. Go to Storage Manager and create M.2 storage pool and volume.
3. In the README it is written as an "extra step" to execute:  `sudo -i <PATH_TO_SCRIPT>/syno_hdd_db.sh --noupdate` to get rid of "drive database outdated" notifications and "unrecognised firmware" warnings, but as of v1.0.6 the changelog states that it is no longer needed. So is this step needed in the latest version (v1.0.7) or not?
4. Schedule the script to run at boot-up. (This is what makes the M.2 volume survives DSM updates. Correct?)

Please inform me if any step is missing or need correction.

Also i would like to understand one more thing. If (for any reason) i want to temporarily revert all changes made by the script i should execute:  `sudo -i <PATH_TO_SCRIPT>/syno_enable_m2_volume.sh --restore`. Then if i want to re-enable the M.2 volumes i should just execute:  `sudo -i <PATH_TO_SCRIPT>/syno_enable_m2_volume.sh` and all the volumes and their shared folders will appear in the file explorer without any corruption?

Once again thanks a lot for your time and work.
Great project.

**Comment by 007revad** [comment, marked as answer]
> * Execute:  `sudo -i <PATH_TO_SCRIPT>/syno_enable_m2_volume.sh`
> * Go to Storage Manager and create M.2 storage pool and volume.

Correct.

> * In the README it is written as an "extra step" to execute:  `sudo -i <PATH_TO_SCRIPT>/syno_hdd_db.sh --noupdate` to get rid of "drive database outdated" notifications and "unrecognised firmware" warnings, but as of v1.0.6 the changelog states that it is no longer needed. So is this step needed in the latest version (v1.0.7) or not?

For some NAS models, before v1.0.6 you needed to run syno_hdd-db.sh to set support_m2_pool="yes". But even with v1.0.6 or v1.0.7 if your NAS model shows annoying warnings in storage manager you can run syno_hdd_db.sh --noupdate to get rid of them.

> * Schedule the script to run at boot-up. (This is what makes the M.2 volume survives DSM updates. Correct?)

Correct.

If you later ran `sudo -i <PATH_TO_SCRIPT>/syno_enable_m2_volume.sh --restore` your NVMe storage pool and volume would still exist and be usable. But you might not be able to manage it from storage manager (repair, rebuild, expand etc).

Depending on your NAS model and DSM version https://github.com/007revad/Synology_HDD_db may be all you need anyway as it makes DSM allow you to create storage pools and volumes on NVMe drives.

**NOTE:** Currently the --restore option in both syno_enable_m2_volume and syno_hdd_db replaces synoinfo.conf with the backed up synoinfo.conf. So if you ran both scripts and then ran 1 of the scripts with the --restore option you'd need to run the other script again without the --restore option.

**Reply by CVasilakis** [reply]
Thanks for your detailed answer.
For a Synology DS720+ running DSM 7.2 and set to auto update DSM versions does the [Synology_HDD_db](https://github.com/007revad/Synology_HDD_db) (scheduled to run at boot-up) be sufficient to avoid any warnings or do i need both [Synology_enable_M2_volume](https://github.com/007revad/Synology_enable_M2_volume) and [Synology_HDD_db](https://github.com/007revad/Synology_HDD_db)?
Thanks again for your time!

**Comment by 007revad** [comment]
For a Synology DS720+ running DSM 7.2 and set to auto update DSM versions does the [Synology_HDD_db](https://github.com/007revad/Synology_HDD_db)
